src/core/mempool.rs

```diff
@@ -1009,17 +1009,25 @@ impl MemPoolDB {
     ///  highest-fee-first order.  This method is interruptable -- in the `settings` struct, the
     ///  caller may choose how long to spend iterating before this method stops.
     ///
-    ///  `todo` returns a boolean representing whether or not to keep iterating.
+    ///  `todo` returns an option to a `TransactionEvent` representing the outcome, or None if we
+    ///  hit an error that wasn't transaction specific.
```

Comment:
Would be better framed as:

```rust
/// `todo` controls the iteration: if it returns `None`, then the mempool iteration will stop.
/// if `todo` returns `Some`, the contents of the return are a `TransactionEvent` representing
/// the outcome of trying to include a given transaction in the block.
```

`None` doesn't always indicate an error occurred: it just indicates that the miner is done assembling the block, and wants to halt the iteration.

Reply:
Can you also comment on how the new `Vec<TransactionEvent>` argument is being used?